de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/concurrent/jupiter/internal/Jupiter.java

```diff
@@ -142,22 +144,17 @@ public boolean isCurrent(Timestamp timestamp) throws TransformationException {
   public Operation receiveJupiterActivity(JupiterActivity jupiterActivity)
       throws TransformationException {
 
-    if (jupiterActivity.getOperation() instanceof TimestampOperation) {
```

Comment:
This got added with commit https://github.com/saros-project/saros/commit/3276dc90c76f0afe356dbbf5f86f8c9cb28c9d9d

You just removed it. Problem is the following. Almost nobody knows how this Jupiter stuff really works and so it is not the best decision to make modifications to files that where not touched since almost 10 years.

Reply:
As you can see [here](https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/concurrent/jupiter/internal/Jupiter.java#L147) and more importantly [here](https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/concurrent/jupiter/internal/text/TimestampOperation.java#L14) or alternatively by just doing `grep -r . TimestampOperation` TimestampOperations are **never** used in the whole codebase.

They were most likely intended to serve a similar purpose, but it was never implemented. So alternatively I could just add a new Operation and afterwards make a cleanup PR that removes the unneeded TimestampOperations, but I do not see the point in that.

> it is not the best decision to make modifications to files that where not touched since almost 10 years.

Yeah it is scary, which is why I spent days testing this. But not touching anything would mean the memory leak never gets fixed and that cannot possibly be your opinion on the subject, right?